aten/src/ATen/core/dispatch/OperatorEntry.h

```diff
@@ -87,13 +94,30 @@ class OperatorEntry final {
   > kernels_;
   std::mutex kernelsMutex_; // protects kernels_
 
+  // unboxedAutogradKernels_ stores all autograd kernels registered for this op.
+  // An autograd kernel has the same signature as the main op kernel and
+  // internally re-dispatches to call the actual kernel.
+  // Autograd kernels are unboxed currently. We are planning to move this
+  // towards a system where ops register autograd wrappers (i.e. functions that
+  // do some wrapping code and get a pointer to the actual kernel) instead of
+  // autograd functions.
+  // This is a list because, similar to kernels_, multiple libraries could
```

Comment:
why this complexity? if this is a temp solution - just keep it simple and have a single pointer. For multiple wrappers we'd need to build something different. But even in that case I think errorring out instead of creating a list is the right behavior

Reply:
it mirrors what we're doing for kernels, i.e. allow overwriting already existing kernels but in a safe way and reset them once the overwriting library is gone. A potential use for this would be XLA overwriting the default autograd kernels provided by ATen.

I think the complexity is worth it - even though this is hopefully a temporary solution, these kind of solutions tend to stick around and cause issues if not done right.

Reply:
as discussed - let's extend this comment to note that we primarily want to enable iPython notebooks. Otherwise, this behavior is actually bad - as you want to get errors, not silent overwrite on collision.

Reply:
I added a comment